### DISCUSSION TITLE
Issue with User Registration and Authentication in Supabase

### DISCUSSION THREAD

**Discussion by baguni99** [supabase/supabase#16410] [Questions]
I'm planning to raise this issue on the Supabase GitHub repository. The problem I'm encountering is that when I click the <button type="submit" disabled={loading}> button to register as a new user, I receive the following error in the console:
`AuthApiError: Database error saving new user
    at handleError (fetch.ts:49:1)
    at async _handleRequest (fetch.ts:128:1)
    at async _request (fetch.ts:95:1)
    at async SupabaseAuthClient.signUp (GoTrueClient.ts:358:1)
    at async joinUsHandler (Register.tsx:29:1)
`
Additionally, the user information is not being stored in Supabase's authentication, and I'm not receiving any confirmation emails. I wanted to share this issue in English on the Supabase GitHub repository to seek assistance and clarification.

`import React, { useState } from "react";
import supabase from "../config/supabaseClient";
import { styled } from "styled-components";

interface FormEvent extends React.FormEvent<HTMLFormElement> {}

const Register: React.FC = () => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [name, setName] = useState("");
  const [confirmPassword, setConfirmPassword] = useState("");
  const [error, setError] = useState("");
  const [loading, setLoading] = useState(false);
  const [successRegister, setSuccessRegister] = useState(false);

  const joinUsHandler = async (e: FormEvent) => {
    e.preventDefault();
    if (!email || !password || !name || !confirmPassword) {
      setError("모든 칸을 입력하세요.");
      return;
    }
    if (password !== confirmPassword) {
      setError("비밀번호가 일치하지 않습니다.");
      return;
    }
    try {
      setLoading(true);

      const signUpResult: any = await supabase.auth.signUp({
        email,
        password,
      });

      if (signUpResult.error) {
        console.error(signUpResult.error);
        if (signUpResult.error.message.includes("Email rate limit exceeded")) {
          alert("이메일 발송 제한이 초과되었습니다. 나중에 다시 시도해주세요.");
        }
        return;
      }

      const user = signUpResult.user;

      if (user) {
        console.log("User registered:", user);

        // Insert user data into "users" table
        await addUser(user.id, name);

        // Insert user data into "profiles" table
        await addProfile(user.id, name);

        setError("");
        console.log("User added to users and profiles tables");
        setSuccessRegister(true);
      }
    } catch (error) {
      console.error(error);
    } finally {
      setLoading(false);
    }
  };

  const addUser = async (userUid: string, userName: string) => {
    const { data, error } = await supabase
      .from("users")
      .upsert([{ user_id: userUid, name: userName }]);

    if (error) {
      console.error(error);
    } else {
      console.log("User added:", data);
    }
  };

  const addProfile = async (userUid: string, userName: string) => {
    const { data, error } = await supabase
      .from("profiles")
      .upsert([{ user_id: userUid, full_name: userName }]);

    if (error) {
      console.error(error);
    } else {
      console.log("Profile added:", data);
    }
  };

  return (
    <CenteredContainer>
      <RegisterContents>
        {successRegister ? (
          <p style={{ color: "green" }}>이메일을 확인해주세요</p>
        ) : (
          <div>
            Join Us !
            <RegistFormContainer onSubmit={joinUsHandler}>
              <input
                type="text"
                value={name}
                onChange={(e) => setName(e.target.value)}
                placeholder="user name"
              />
              <input
                type="email"
                value={email}
                onChange={(e) => setEmail(e.target.value)}
                placeholder="email"
                autoComplete="username"
              />
              <input
                type="password"
                value={password}
                onChange={(e) => setPassword(e.target.value)}
                placeholder="password"
                autoComplete="new-password"
              />
              <input
                type="password"
                value={confirmPassword}
                onChange={(e) => setConfirmPassword(e.target.value)}
                placeholder="비밀번호 확인"
                autoComplete="new-password"
              />
              <button type="submit" disabled={loading}>
                {loading ? "회원가입 중..." : "회원가입 완료"}
              </button>
              {error && <p style={{ color: "red" }}>{error}</p>}
            </RegistFormContainer>
          </div>
        )}
      </RegisterContents>
    </CenteredContainer>
  );
};

const CenteredContainer = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100vh;
`;

const RegisterContents = styled.div`
  border: 1px solid lightgray;
  padding: 20px;
  border-radius: 20px;
  width: 400px;
  height: 300px;
`;

const RegistFormContainer = styled.form`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: 100%;
`;

export default Register;
`

and this is my full code

**Comment by GaryAustin1** [comment]
Database error saving user is almost always having a trigger function on auth.users to populate a public table getting an error.
You can look in the dashboard Postgres logs for more info if it is a database error caused by that process.
Do you have such a trigger function, possibly from the user management starter guides?
https://github.com/orgs/supabase/discussions/13043

**Reply by GaryAustin1** [reply]
If you are getting the same error, that further points to the possibility I mentioned.
Did you look at the logs, check for an auth.users insert function that you might be using?  Look at the link?

Need to rule that out to determine if it is something wrong with your instance itself.

**Comment by baguni99** [comment]
Hello,
Upon my investigation, I've come to realize that it's not possible to add users by clicking "Add Users" in the table, and it seems that registering new members altogether is also not feasible. This might indicate an issue with the dashboard or the settings within the auth or table sections. Could you provide guidance on how this can be resolved?

**Reply by baguni99** [reply]
hi I saw that url and I found my error message

Log ID
13a21324-d798-45f0-8603-845c914b3fc2

Log Timestamp (UTC)
2023-08-10T16:36:40.000Z

Log Event Message
{"component":"api","error":"failed to close prepared statement: ERROR: current transaction is aborted, commands ignored until end of transaction block (SQLSTATE 25P02): ERROR: column \"id\" of relation \"profiles\" does not exist (SQLSTATE 42703)","level":"error","method":"POST","msg":"500: Database error creating new user","path":"/admin/users","referer":"https://supabase.com/dashboard/project/pqpkkqfpyvcbuhowcqaf/auth/users","remote_addr":"58.72.47.252","time":"2023-08-10T16:36:40Z","timestamp":"2023-08-10T16:36:39Z"}

Log Metadata
[
  {
    "message": null,
    "timestamp": "2023-08-10T16:36:39Z",
    "__MONOTONIC_TIMESTAMP": null,
    "CODE_FUNC": null,
    "instance_id": null,
    "status": null,
    "_CMDLINE": null,
    "method": "POST",
    "_SYSTEMD_CGROUP": null,
    "CODE_FILE": null,
    "EXECUTABLE": null,
    "_EXE": null,
    "UNIT": null,
    "level": "error",
    "_COMM": null,
    "duration": null,
    "issuer": null,
    "_LINE_BREAK": null,
    "_SOURCE_REALTIME_TIMESTAMP": null,
    "msg": "500: Database error creating new user",
    "action": null,
    "login_method": null,
    "_UID": null,
    "host": "db-pqpkkqfpyvcbuhowcqaf",
    "PRIORITY": null,
    "_CAP_EFFECTIVE": null,
    "_PID": null,
    "INVOCATION_ID": null,
    "_SYSTEMD_UNIT": null,
    "source_type": null,
    "SYSLOG_FACILITY": null,
    "request_id": null,
    "CODE_LINE": null,
    "path": "/admin/users",
    "component": "api",
    "project": null,
    "user_id": null,
    "auth_event": [],
    "args": [],
    "referer": "https://supabase.com/dashboard/project/pqpkkqfpyvcbuhowcqaf/auth/users",
    "factor_id": null,
    "provider": null,
    "client_id": null,
    "remote_addr": "58.72.47.252",
    "_SYSTEMD_SLICE": null,
    "_SYSTEMD_INVOCATION_ID": null,
    "header": null,
    "_MACHINE_ID": null,
    "_AUDIT_LOGINUID": null,
    "_TRANSPORT": null,
    "_SELINUX_CONTEXT": null,
    "MESSAGE_ID": null,
    "__REALTIME_TIMESTAMP": null,
    "metadata": [],
    "_STREAM_ID": null,
    "metering": null,
    "time": null,
    "_GID": null,
    "_BOOT_ID": null,
    "SYSLOG_IDENTIFIER": null,
    "_AUDIT_SESSION": null,
    "error": "failed to close prepared statement: ERROR: current transaction is aborted, commands ignored until end of transaction block (SQLSTATE 25P02): ERROR: column \"id\" of relation \"profiles\" does not exist (SQLSTATE 42703)"
  }
]

**Reply by GaryAustin1** [reply]
So you have a user management style trigger function and a profiles table.  Your function code does not match your profiles table.  The function code wants to insert to a column called id, but your profiles table does not have that column.  You have to change the function code or change the table.
